homeassistant/components/flick_electric/sensor.py

```diff
@@ -39,36 +36,22 @@ class FlickPricingSensor(SensorEntity):
     _attr_translation_key = "power_price"
     _attributes: dict[str, Any] = {}
 
-    def __init__(self, api: FlickAPI) -> None:
+    def __init__(self, coordinator: FlickElectricDataCoordinator) -> None:
         """Entity object for Flick Electric sensor."""
-        self._api: FlickAPI = api
-        self._price: FlickPrice = None
+        super().__init__(coordinator)
+
+        self._attr_unique_id = f"{DOMAIN}_{coordinator.supply_node_ref}_pricing"
 
     @property
-    def native_value(self):
+    def native_value(self) -> Decimal:
         """Return the state of the sensor."""
-        return self._price.price
+        return self.coordinator.data.price
 
     @property
-    def extra_state_attributes(self):
+    def extra_state_attributes(self) -> dict[str, Any] | None:
         """Return the state attributes."""
-        return self._attributes
-
-    async def async_update(self) -> None:
-        """Get the Flick Pricing data from the web service."""
-        if self._price and self._price.end_at >= utcnow():
-            return  # Power price data is still valid
-
-        async with asyncio.timeout(60):
-            self._price = await self._api.getPricing()
-
-        _LOGGER.debug("Pricing data: %s", self._price)
-
-        self._attributes[ATTR_START_AT] = self._price.start_at
-        self._attributes[ATTR_END_AT] = self._price.end_at
-        for component in self._price.components:
-            if component.charge_setter not in ATTR_COMPONENTS:
-                _LOGGER.warning("Found unknown component: %s", component.charge_setter)
-                continue
-
-            self._attributes[component.charge_setter] = float(component.value)
+        return {
+            ATTR_START_AT: self.coordinator.data.start_at,
+            ATTR_END_AT: self.coordinator.data.end_at,
+            # TODO: Components
+        }
```

Comment:
Another remark that's not for this PR, but ideally we try to split out attributes into their own entities or use action responses for things that return large amounts of data.